programs/pluto/ipsec_interface.c

```diff
@@ -452,7 +453,12 @@ static struct ipsec_interface *alloc_ipsec_interface(ipsec_interface_id_t ipsec_
 			 kernel_ops->ipsec_interface->name,
 			 unmap_id(ipsec_if_id));
 	passert(l < IFNAMSIZ);
-	jam_str(p->physical, sizeof(p->physical), iface->real_device_name);
+	if (kernel_xfrmi_req_phy()) {
```

Comment:
Needing to specify the physical device when creating an ipsec interface seems to be a Linux only thing (going by the `ipsec-interface-00-*` tests).  Now that Linux has dropped the reqirement, why not just drop the suffix ('19 is a long time ago).

Reply:
only output change  would be  pluto log lines? "created ipsec-interface ipsec21@NONE"  "delete", while  "ip link" output would still have NONE. So I guess it is not great idea, otherwise we would have sanitize that NONE

Reply:
Yes, the output would change to `created ipsec-interface ipsec21` et.al.  Does `ip link` really show ipsec21@NONE!

Reply:
So @... can be dropped from `alloc_ipsec_interface()`  (and `ip link` just sounds broken).
Just one thing to check, if `ipsec21@eth1` is specified, does it do anything or is it ignored?